src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_xml.rs

```diff
@@ -0,0 +1,257 @@
+use std::collections::HashMap;
+//  Copyright 2022 Datafuse Labs.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+use std::sync::Arc;
+
+use common_datavalues::DataSchemaRef;
+use common_datavalues::TypeDeserializer;
+use common_datavalues::TypeDeserializerImpl;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_io::prelude::FormatSettings;
+use common_io::prelude::NestedCheckpointReader;
+use common_meta_types::StageFileFormatType;
+use common_settings::Settings;
+use xml::reader::XmlEvent;
+use xml::ParserConfig;
+
+use crate::processors::sources::input_formats::impls::input_format_tsv::format_column_error;
+use crate::processors::sources::input_formats::input_format_text::get_time_zone;
+use crate::processors::sources::input_formats::input_format_text::AligningState;
+use crate::processors::sources::input_formats::input_format_text::BlockBuilder;
+use crate::processors::sources::input_formats::input_format_text::InputFormatTextBase;
+use crate::processors::sources::input_formats::input_format_text::RowBatch;
+use crate::processors::sources::input_formats::InputContext;
+
+pub struct InputFormatXML {}
+
+impl InputFormatXML {
+    fn read_row(
+        buf: &[u8],
+        deserializers: &mut [TypeDeserializerImpl],
+        schema: &DataSchemaRef,
+        format_settings: &FormatSettings,
+        path: &str,
+        row_index: usize,
+    ) -> Result<()> {
+        let mut row_json: serde_json::Value = serde_json::from_reader(buf)?;
```

Comment:
What's this for? We need to read XML as json first?

Reply:
I guess this pr is not ready for review  @eliasyaoyc? maybe you can convert this pr a draft?

Reply:
```
let cols_bytes = serde_json::to_vec(&cols)?;

output.data.extend_from_slice(&cols_bytes);
```
since i converted each row of data to json, i originally used Vec<u8>, but in order to solve the problem of inconsistent column order  in xml and table order , i used map to solve this problem, so i conveted it to json for read_row to read easy

Reply:
not familiar with the XML lib you use, 
if it does not support streaming in(bytes) and out(XMLEvent),  I suggest to use Aligner only to accumulate and concat all ReadBatch, and sent all content of the File a RowBatch and call XML reader there.